suricata/update/main.py

```diff
@@ -1264,6 +1267,12 @@ def _main():
         logger.info("Loading %s.", drop_conf_filename)
         drop_filters += load_drop_filters(drop_conf_filename)
 
+    # Load user provided output rules filename
+    rule_filename = config.get("output-rule-filename")
+    if rule_filename:
+        logger.info("Setting output rule filename to %s", rule_filename)
+        DEFAULT_OUTPUT_RULE_FILENAME = rule_filename
+
```

Comment:
If we try to examine this. This will lead our following code using the `DEFAULT_OUTPUT_RULE_FILENAME` variable to break. See below.
```
└─ $ ▶ ./bin/suricata-update
....
8/11/2018 -- 21:30:17 - <Info> -- Backing up current rules.
Traceback (most recent call last):
  File "./bin/suricata-update", line 33, in <module>
    sys.exit(main.main())
  File "/home/unixia/suricata/suricata-update/suricata/update/main.py", line 1450, in main
    sys.exit(_main())
  File "/home/unixia/suricata/suricata-update/suricata/update/main.py", line 1399, in _main
    config.get_output_dir(), DEFAULT_OUTPUT_RULE_FILENAME)
UnboundLocalError: local variable 'DEFAULT_OUTPUT_RULE_FILENAME' referenced before assignment
```
Since you have used an assignment statement at line 1274, the following line at 1399 tends to fail in case the control goes there. To read in detail why it happened, read [this](https://docs.python.org/3/faq/programming.html#why-am-i-getting-an-unboundlocalerror-when-the-variable-has-a-value).

Please test the various cases that might be there for running the command.